src/calendar/interfaces.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { NonCancelableEventHandler } from '../internal/events';
+
+export interface CalendarProps {
+  /**
+   * The current input value, in YYYY-MM-DD format.
+   */
+  value: string;
+
+  /**
+   * Defines whether a particular date is enabled in the calendar or not.
+   * If you disable a date in the calendar, users can still enter this date using a keyboard.
+   * We recommend that you also validate these constraints on the client-side and server-side
+   * as you would for other form elements.
+   * @param date
+   */
+  isDateEnabled?: CalendarProps.IsDateEnabledFunction;
+
+  /**
+   * Specifies the locale to use to render month names and determine the starting day of the week.
```

Comment:
Also the day for the week? 
What do you mean by starting day of the week? What is the difference to `startOfWeek`